src/smart.ts

```diff
@@ -233,10 +233,11 @@ export async function authorize(
 
     const storage = env.getStorage();
 
-    // For these three an url param takes precedence over inline option
+    // For these four an url param takes precedence over inline option
     iss            = url.searchParams.get("iss")            || iss;
     fhirServiceUrl = url.searchParams.get("fhirServiceUrl") || fhirServiceUrl;
     launch         = url.searchParams.get("launch")         || launch;
+    patientId      = url.searchParams.get("patientId")      || patientId;
```

Comment:
`patientId` is a constant, so it can't be reassigned here. I think you only need to move its initialization from line 215 down to the `let` variables section (223 to 232). If you do that this PR should be ready for merging.